cmd/otelcol/main.go

```diff
@@ -164,44 +156,72 @@ func checkRuntimeParams() {
 	// Check if memory ballast flag was passed
 	// If so, ensure memory ballast env var is not set
 	// Then set memory ballast and limit properly
-	ballastSize := getKeyValue(args, "--mem-ballast-size-mib")
+	ballastSize := getKeyValue(os.Args[1:], "--mem-ballast-size-mib")
 	if ballastSize != "" {
-		config = os.Getenv(ballastEnvVarName)
-		if config != "" {
-			log.Fatalf("Both %v and '--config' were specified, but only one is allowed", ballastEnvVarName)
+		if os.Getenv(ballastEnvVarName) != "" {
+			log.Fatalf("Both %v and '--mem-ballast-size-mib' were specified, but only one is allowed", ballastEnvVarName)
 		}
 		os.Setenv(ballastEnvVarName, ballastSize)
 	}
 	setMemoryBallast(memTotalSizeMiB)
 	setMemoryLimit(memTotalSizeMiB)
 }
 
-// Validate and set the configuration
-func setConfig() {
-	// Check if the config is specified via the env var.
-	config := os.Getenv(configEnvVarName)
-	// If not attempt to use a default config; supports Docker and local
-	if config == "" {
+// Validate and equate specified config file path flag to the config file path env var
+func setConfigSource() {
+	// Config file path from cmd flag --config.
+	pathFlag := getKeyValue(os.Args[1:], "--config")
+	// Config file path from env var.
+	pathVar := os.Getenv(configEnvVarName)
+	// Config YAML from env var.
+	yamlVar := os.Getenv(configYamlEnvVarName)
+
+	// Config file path flag `--config` should take priority when running from most contexts.
+	if pathFlag != "" {
+		// Config file path flag takes precedence over config YAML env var.
+		if yamlVar != "" {
+			log.Printf("Both environment variable %v and flag '--config' were specified. Ignoring configuration in the environment variable and using configuration in flag value %q", configYamlEnvVarName, pathFlag)
+		}
+		// Config file path flag takes precedence over config file path env var.
+		if pathVar != "" && pathVar != pathFlag {
+			log.Printf("Both environment variable %v and flag '--config' were specified. Overriding %q environment variable value with flag value %q for this session", configEnvVarName, pathVar, pathFlag)
+		}
+		// Setting the config file path env var to the config file path flag value.
+		pathVar = pathFlag
+		os.Setenv(configEnvVarName, pathVar)
+	} else {
+		// Restricting specifying config file path and config YAML env vars simultaneously.
+		if pathVar != "" && yamlVar != "" {
+			log.Fatalf("Specifying env vars %s and %s simultaneously is not allowed", configEnvVarName, configYamlEnvVarName)
```

Comment:
I think `SPLUNK_CONFIG` should take priority so that `SPLUNK_CONFIG_YAML` will only be used if no path is specified.